legend-engine-core/legend-engine-core-base/legend-engine-core-language-pure/legend-engine-language-pure-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/to/PureGrammarComposerUtility.java

```diff
@@ -150,7 +150,7 @@ public static String convertIdentifier(String val, boolean doubleQuotes)
         }
         else
         {
-            return UNQUOTED_IDENTIFIER_PATTERN.matcher(val).matches() ? val : convertString(val, true, doubleQuotes);
+            return convertString(val, true, doubleQuotes);
```

Comment:
This is not a good change, as will trigger many breaks on test cases, and will impact users that dont need quote.

I though the change is either to enhance the regex, or enhance the grammar definition so we can parse the value properly.